### PR TITLE
[13.0][ADD] storage_backend_ftp

### PR DESCRIPTION
**Pull request by acsonefho**
New module `storage_backend_ftp`

**Note:** I didn't test every security protocols

### CODE REVIEW

File: storage_backend_ftp/components/ftp_adapter.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                raise FileNotFoundError(repr(e))
+            buff.seek(0)
+            data = buff.read()
+        return data.encode()
+
+    def list(self, relative_path):
+        full_path = self._fullpath(relative_path)
+        with ftp(self.collection) as client:
+            try:
+                return client.retrlines("LIST " + full_path)
```
Comment (LoisRForgeFlow) [inline storage_backend_ftp/components/ftp_adapter.py]: This does not return a list. But a string with a command-like message. I think you should use `client.nlst(full_path)`

**Review by LoisRForgeFlow — COMMENTED** [review]
I did an initial test, see a comment inline.

**Comment by LoisRForgeFlow** [comment]
Hi @acsonefho, I have done a full testing and found a couple of issues. I have created a PR to your branch addressing them, could you have a look? https://github.com/acsone/storage/pull/1

File: storage_backend_ftp/components/ftp_adapter.py (hunk 1 above)
Comment (LoisRForgeFlow) [reply storage_backend_ftp/components/ftp_adapter.py]: addressed in https://github.com/acsone/storage/pull/1.

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

File: storage_backend_ftp/components/ftp_adapter.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    params = {}
+    security = None
+    if backend.ftp_encryption == "ftp":
+        ftp = ftplib.FTP
+    elif backend.ftp_encryption == "tls":
+        ftp = ftplib.FTP_TLS
+        # Due to a bug into between ftplib and ssl, this part (about ssl) might not work!
+        # https://bugs.python.org/issue31727
+        security = None
+        if backend.ftp_security == "tls":
```
Comment (simahawk) [inline storage_backend_ftp/components/ftp_adapter.py:43]: pls use a dict :wink:

File: storage_backend_ftp/tests/test_ftp.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        mocked_filepaths = bad_filepaths + good_filepaths
+        backend = self.backend.sudo()
+        expected = good_filepaths[:]
+        expected = [backend.directory_path + "/" + path for path in good_filepaths]
+        self._test_find_files(
+            backend, ADAPTER_PATH, mocked_filepaths, r".*\.good$", expected
+        )
+
+    # Do not patch the entire ftplib otherwise the error_perm Exception
+    # become also a mock and then a traceback is genrated on the "except ftplib.error_perm"
```
Comment (simahawk) [inline storage_backend_ftp/tests/test_ftp.py:110]:
```suggestion
    # become also a mock and then a traceback is generated on the "except ftplib.error_perm"
```

File: storage_backend_sftp/models/storage_backend.py [hunk 4]
```diff
@@ -12,15 +12,17 @@ class StorageBackend(models.Model):
 
     backend_type = fields.Selection(selection_add=[("sftp", "SFTP")])
     sftp_server = fields.Char(string="SFTP Host")
-    sftp_port = fields.Integer(string="Port", default=22)
+    sftp_port = fields.Integer(string="SFTP Port", default=22)
     sftp_auth_method = fields.Selection(
-        string="Authentification Method",
+        string="SFTP Authentification Method",
```
Comment (simahawk) [inline storage_backend_sftp/models/storage_backend.py:17]:
```suggestion
        string="SFTP Authentication Method",
```

**Review by simahawk — APPROVED** [review]
LG

**Comment by simahawk** [comment]
build is red

**Comment by LoisRForgeFlow** [comment]
@acsonefho could you attend the final comments from @simahawk ?

**Comment by LoisRForgeFlow** [comment]
I have superseded this PR due to inactivity here https://github.com/OCA/storage/pull/134, I think we can close this one.

**Pull request closed without merging**